File: lib-database-entities/src/main/java/com/ustadmobile/lib/db/entities/SaleProductPicture.java
package com.ustadmobile.lib.db.entities;

import com.ustadmobile.lib.database.annotation.UmEntity;
import com.ustadmobile.lib.database.annotation.UmPrimaryKey;
import com.ustadmobile.lib.database.annotation.UmSyncLastChangedBy;
import com.ustadmobile.lib.database.annotation.UmSyncLocalChangeSeqNum;
import com.ustadmobile.lib.database.annotation.UmSyncMasterChangeSeqNum;

@UmEntity(tableId = 73)
public class SaleProductPicture {

    @UmPrimaryKey(autoGenerateSyncable = true)
    private long saleProductPictureUid;

    //The product uid
    private long saleProductPictureSaleProductUid;

    //file size of product picture
    private int saleProductPictureFileSize;

    //picture file's timestamp
    private long saleProductPictureTimestamp;

    //picture file's mime type
    private String saleProductPictureMime;

    @UmSyncMasterChangeSeqNum
    private long saleProductPictureMCSN;

    @UmSyncLocalChangeSeqNum
    private long saleProductPictureLCSN;

    @UmSyncLastChangedBy
    private int saleProductPictureLCB;


    public long getSaleProductPictureUid() {
        return saleProductPictureUid;
    }

    public void setSaleProductPictureUid(long saleProductPictureUid) {
        this.saleProductPictureUid = saleProductPictureUid;
    }

    public long getSaleProductPictureSaleProductUid() {
        return saleProductPictureSaleProductUid;
    }

    public void setSaleProductPictureSaleProductUid(long saleProductPictureSaleProductUid) {
        this.saleProductPictureSaleProductUid = saleProductPictureSaleProductUid;
    }

    public int getSaleProductPictureFileSize() {
        return saleProductPictureFileSize;
    }

    public void setSaleProductPictureFileSize(int saleProductPictureFileSize) {
        this.saleProductPictureFileSize = saleProductPictureFileSize;
    }

    public long getSaleProductPictureTimestamp() {
        return saleProductPictureTimestamp;
    }

    public void setSaleProductPictureTimestamp(long saleProductPictureTimestamp) {
        this.saleProductPictureTimestamp = saleProductPictureTimestamp;
    }

    public String getSaleProductPictureMime() {
        return saleProductPictureMime;
    }

    public void setSaleProductPictureMime(String saleProductPictureMime) {
        this.saleProductPictureMime = saleProductPictureMime;
    }

    public long getSaleProductPictureMCSN() {
        return saleProductPictureMCSN;
    }

    public void setSaleProductPictureMCSN(long saleProductPictureMCSN) {
        this.saleProductPictureMCSN = saleProductPictureMCSN;
    }

    public long getSaleProductPictureLCSN() {
        return saleProductPictureLCSN;
    }

    public void setSaleProductPictureLCSN(long saleProductPictureLCSN) {
        this.saleProductPictureLCSN = saleProductPictureLCSN;
    }

    public int getSaleProductPictureLCB() {
        return saleProductPictureLCB;
    }

    public void setSaleProductPictureLCB(int saleProductPictureLCB) {
        this.saleProductPictureLCB = saleProductPictureLCB;
    }
}

File: lib-database-entities/src/commonMain/kotlin/com/ustadmobile/lib/db/entities/SaleProductPicture.kt
package com.ustadmobile.lib.db.entities

import androidx.room.Entity
import androidx.room.PrimaryKey
import com.ustadmobile.door.annotation.*
import kotlinx.serialization.Serializable

@SyncableEntity(tableId = 73)
@Entity
@Serializable
@EntityWithAttachment
open class SaleProductPicture() {

    @PrimaryKey(autoGenerate = true)
    var saleProductPictureUid: Long = 0

    //The product uid
    var saleProductPictureSaleProductUid: Long = 0

    //file size of product picture
    var saleProductPictureFileSize: Int = 0

    //picture file's timestamp
    var saleProductPictureTimestamp: Long = 0

    //picture file's mime type
    var saleProductPictureMime: String? = null

    @MasterChangeSeqNum
    var saleProductPictureMCSN: Long = 0

    @LocalChangeSeqNum
    var saleProductPictureLCSN: Long = 0

    @LastChangedBy
    var saleProductPictureLCB: Int = 0

    //Index for which picture. Default to zero
    var saleProductPictureIndex : Int = 0

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || this::class != other::class) return false

        other as SaleProductPicture

        if (saleProductPictureUid != other.saleProductPictureUid) return false
        if (saleProductPictureSaleProductUid != other.saleProductPictureSaleProductUid) return false
        if (saleProductPictureFileSize != other.saleProductPictureFileSize) return false
        if (saleProductPictureTimestamp != other.saleProductPictureTimestamp) return false
        if (saleProductPictureMime != other.saleProductPictureMime) return false
        if (saleProductPictureMCSN != other.saleProductPictureMCSN) return false
        if (saleProductPictureLCSN != other.saleProductPictureLCSN) return false
        if (saleProductPictureLCB != other.saleProductPictureLCB) return false
        if (saleProductPictureIndex != other.saleProductPictureIndex) return false

        return true
    }

    override fun hashCode(): Int {
        var result = saleProductPictureUid.hashCode()
        result = 31 * result + saleProductPictureSaleProductUid.hashCode()
        result = 31 * result + saleProductPictureFileSize
        result = 31 * result + saleProductPictureTimestamp.hashCode()
        result = 31 * result + (saleProductPictureMime?.hashCode() ?: 0)
        result = 31 * result + saleProductPictureMCSN.hashCode()
        result = 31 * result + saleProductPictureLCSN.hashCode()
        result = 31 * result + saleProductPictureLCB
        result = 31 * result + saleProductPictureIndex
        return result
    }


}
